### PR TITLE
update node version from 0.12.7 to 4.4.0 (LTS)

### LINKED ISSUE #442: Switch from Node 0.12 to Node 4.x?
Re: [#436 (comment)](https://github.com/mozilla/testpilot/pull/436/files#r54643575) thread about Node 0.12 usage (in Dockerfile and Travis-CI),

Not sure if it's worth it to just switch to Node 4.x now since it has a much friendlier LTS+maintenance window (see the handy table and chart at the bottom of https://github.com/nodejs/LTS#lts_schedule), and I'm sure it will be a lot easier to bump minor Node 4.x versions in the future versus trying to jump from Node 0.12 to Node 4 while trying to solve future SSL exploits.
1. [/.travis.yml:13](https://github.com/mozilla/testpilot/blob/1425cf9b5b68a7f06b69c0cb8708d0c86635bc6a/.travis.yml#L13) &mdash; `nvm install 0.12`
2. [/Dockerfile-frontend-build:1](https://github.com/mozilla/testpilot/blob/1425cf9b5b68a7f06b69c0cb8708d0c86635bc6a/Dockerfile-frontend-build#L1) &mdash; `FROM node:0.12.7`

(Plus I can't find Node 0.12 listed on the [nodejs homepage](https://nodejs.org/en/))

### PR DESCRIPTION
**Pull request by meandavejustice**
-fixes #442

### CODE REVIEW

File: addon/package.json [hunk 1]
```diff
@@ -13,7 +13,8 @@
   "author": "Mozilla (https://mozilla.org/)",
   "engines": {
     "firefox": ">=38.0a1",
-    "fennec": ">=38.0a1"
+    "fennec": ">=38.0a1",
+    "node" : ">=0.4.0"
```
Comment (lmorchard) [inline addon/package.json]: Looks like Circle CI tests failed on this - I don't think that's a valid engines entry for the addon

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "bugs": {
     "url": "https://github.com/mozilla/testpilot/issues"
   },
+  "engines" : { "node" : ">=0.4.0" },
```
Comment (lmorchard) [inline package.json]: Should this be 4.4.0?

Comment (meandavejustice) [inline package.json]: woah, yes, I'll fix that rn

**Comment by lmorchard** [comment]
Fixed the failure in Circle CI by re-building without a cache. Looks like upgrading the node version means we have to blow away node_modules and re-run `npm install` from scratch. Not entirely surprising.

**Pull request merged**